lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -123,25 +121,26 @@ function dosomething_reportback_get_reportbacks_query($params) {
   return $result;
 }
 
-
 /**
- * Filter out requested flagged Reportbacks based on user permissions.
- *
- * @param  array  $params
- * @return mixed
+ * Determine if user has access to flagged Reportbacks based on user permission. 
+ * @param array $results
  */
-function dosomething_reportback_filter_flagged_reportbacks($params) {
+function dosomething_reportback_accessible_results ($results) {
   if (user_access('view any reportback')) {
-    if (isset($params['flagged'])) {
-      unset($params['flagged']);
+    return $results;
+  }
+  else {
+    foreach ($results as $key => $reportback) {
+      if ($reportback->flagged == 1) {
```

Comment:
Hmmm, just maybe confirm it works with a string of `"1"` as well. It might just be for string `"true"`.

If not, then maybe you can update the helper function to handle both a string "true/false" and a string "0/1" :wink: 

Would definitely come in handy!